src/hermes/commands/deposit/invenio.py

```diff
@@ -17,648 +17,613 @@
 
-    # Store link to latest draft to be used in :func:`update_metadata`.
-    old_deposit = response.json()
-    ctx.update(invenio_path["links"]["latestDraft"], old_deposit['links']['latest_draft'])
+        :param record_id: The record that sould be resolved.
+        :return: The record id of the latest version for the requested record.
+        """
+        res = self.client.get_record(record_id)
+        if res.status_code != 200:
+            raise ValueError(f"Could not retrieve record from {res.url}: {res.text}")
 
+        res_json = res.json()
+        res = self.client.get(res_json['links']['latest'])
+        if res.status_code != 200:
+            raise ValueError(f"Could not retrieve record from {res.url}: {res.text}")
 
-def update_metadata(click_ctx: click.Context, ctx: CodeMetaContext):
-    """Update the metadata of a draft.
+        res_json = res.json()
+        return res_json['id'], res_json['metadata']
 
-    If no draft is found in the context, it is assumed that no metadata has to be
-    updated (e.g. because an initial version was created already containing the
-    metadata).
-    """
-
-    invenio_path = ContextPath.parse("deposit.invenio")
-    invenio_ctx = ctx[invenio_path]
-    draft_url = invenio_ctx.get("links", {}).get("latestDraft")
-
-    if draft_url is None:
-        return
 
-    _log = logging.getLogger("cli.deposit.invenio")
-
-    deposition_metadata = invenio_ctx["depositionMetadata"]
+class InvenioDepositPlugin(BaseDepositPlugin):
 
-    response = requests.put(
-        draft_url,
-        json={"metadata": deposition_metadata},
-        headers={
-            "User-Agent": hermes_user_agent,
-            "Authorization": f"Bearer {click_ctx.params['auth_token']}",
-        }
-    )
-
-    if not response.ok:
-        raise RuntimeError(f"Could not update metadata of draft {draft_url!r}")
+    platform_name = "invenio"
+    invenio_client_class = InvenioClient
+    invenio_resolver_class = InvenioResolver
 
-    deposit = response.json()
-    _log.debug("Created new version deposit: %s", deposit["links"]["html"])
-    with open(ctx.get_cache('deposit', 'deposit', create=True), 'w') as deposit_file:
-        json.dump(deposit, deposit_file, indent=4)
-
-    ctx.update(invenio_path["links"]["bucket"], deposit["links"]["bucket"])
-    ctx.update(invenio_path["links"]["publish"], deposit["links"]["publish"])
+    def __init__(self, click_ctx: click.Context, ctx: CodeMetaContext, client=None, resolver=None) -> None:
+        super().__init__(click_ctx, ctx)
 
+        self.invenio_context_path = ContextPath.parse(f"deposit.{self.platform_name}")
+        self.invenio_ctx = None
 
-def delete_artifacts(click_ctx: click.Context, ctx: CodeMetaContext):
-    """Delete existing file artifacts.
+        auth_token = self.click_ctx.params.get("auth_token")
+        if auth_token is None:
+            raise DepositionUnauthorizedError("No auth token given for deposition platform")
```

Comment:
**chore:** This also fails if we inject a fully configured client. I think this should not be the case.